activerecord/test/models/sharded.rb

```diff
@@ -5,3 +5,5 @@
 require_relative "sharded/comment"
 require_relative "sharded/tag"
 require_relative "sharded/blog_post_tag"
+require_relative "sharded/blog_post_destroy_async"
+require_relative "sharded/comment_destroy_async"
```

Comment:
Ah found it. This is what's causing Active Record to load the Active Job constant. We can't require these here, we have to only require them when needed in the async active job tests only.